Makefile

```diff
@@ -295,22 +295,38 @@ check: generate fmt vet docs golangci-lint imports ## Dev: Run code checks (go f
 	make generate manifests -C pkg/plugins/resources/k8s/native
 	git diff --quiet || test $$(git diff --name-only | grep -v -e 'go.mod$$' -e 'go.sum$$' | wc -l) -eq 0 || ( echo "The following changes (result of code generators and code checks) have been detected:" && git --no-pager diff && false ) # fail if Git working tree is dirty
 
-test: ## Dev: Run tests
+test: ${COVERAGE_PROFILE} test/api test/k8s test/kuma coverage ## Dev: Run tests for all modules
+
+${COVERAGE_PROFILE}:
 	mkdir -p "$(shell dirname "$(COVERAGE_PROFILE)")"
-	$(GO_TEST) $(GO_TEST_OPTS) -race -covermode=atomic -coverpkg=./... -coverprofile="$(COVERAGE_PROFILE)" $(PKG_LIST)
+
+coverage: ${COVERAGE_PROFILE}
 	go tool cover -html="$(COVERAGE_PROFILE)" -o "$(COVERAGE_REPORT_HTML)"
 
+test/kuma: # Dev: Run tests for the module github.com/Kong/kuma
+	$(GO_TEST) $(GO_TEST_OPTS) -race -covermode=atomic -coverpkg=./... -coverprofile="$(COVERAGE_PROFILE)" $(PKG_LIST)
+
+test/api: # Dev: Run tests for the module github.com/Kong/kuma/api
+	GO_TEST='${GO_TEST}' GO_TEST_OPTS='${GO_TEST_OPTS}' COVERAGE_PROFILE='../$(BUILD_COVERAGE_DIR)/coverage-api.out' \
+	make test -C ./api
+
+test/k8s: # Dev: Run tests for the module github.com/Kong/kuma/pkg/plugins/resources/k8s/native
+	GO_TEST='${GO_TEST}' GO_TEST_OPTS='${GO_TEST_OPTS}' COVERAGE_PROFILE='../../../../../$(BUILD_COVERAGE_DIR)/coverage-k8s.out' \
+	make test -C ./pkg/plugins/resources/k8s/native
+
```

Comment:
Hmm can we have a common target that gets arguments for PKG and COVERAGE to test and input the results to?

Reply:
Are you talking about something like this:
```Makefile
test/api: MODULE=./api COVERAGE_PROFILE=../$(BUILD_COVERAGE_DIR)/coverage-api.out
test/api: test/module

test/k8s: MODULE=./pkg/plugins/resources/k8s/native COVERAGE_PROFILE=../../../../../$(BUILD_COVERAGE_DIR)/coverage-k8s.out
test/k8s: test/module

test/module:
	GO_TEST='${GO_TEST}' GO_TEST_OPTS='${GO_TEST_OPTS}' COVERAGE_PROFILE='${COVERAGE_PROFILE}' make test -C ${MODULE}
```

Reply:
Yes! At least we get a more structured testing pattern to follow. @jakubdyszkiewicz what do you think?

Reply:
Sure, as long as we've got a single `make test` target that tests everything (which I see we do) I'm ok with it